app/models/prisoner_step.rb

```diff
@@ -16,6 +16,6 @@ class PrisonerStep
   delegate :name, to: :prison, prefix: true
 
   def prison
-    Prison.find_by(id: prison_id)
+    prison_id ? Prison.find_by_id(prison_id) : nil
```

Comment:
Yes, arguably we should use an explicitly different API to avoid mistaking it for an ActiveRecord find. I'll keep that in mind.